tests/components/mqtt/test_common.py

```diff
@@ -1573,7 +1572,7 @@ async def help_test_publishing_with_custom_encoding(
         )
         if test_data["encoding"] is not None:
             test_config_setup["encoding"] = test_data["encoding"]
-        if test_data["cmd_tpl"]:
+        if template and test_data["cmd_tpl"]:
```

Comment:
Why did this need changing?

Reply:
It seemed the test entities were not set up, since this fails now with the new config the tests failed as well.
Some testcases include testing with a template, not all. In test cases that do not include a template this test needs to be skipped.